open-craft/xblock-lti-consumer
# BB-2250: LTI 1.3 proof of concept implementation

This PR implements a proof-of-concept implementation of the LTI 1.3 standard to the current LTI Block without breaking LTI 1.1 support.

**Setup instructions:**
1. Clone this branch on your devstack's `src` folder 
2. Install in LMS and Studio 
```
$ lms-shell
> pip install -e /edx/src/xblock-lti-consumer
```
3. Set up a local tunnel (using `ngrok` or a similar tool) 
4. Set up a tool in the IMS Global reference implementation (https://lti-ri.imsglobal.org/lti/tools/).

**Testing instructions:**
1. Create a new course, and add the `lti_consumer` block to the advanced modules list.
2. In the course, create a new unit and add the LTI block.
3. In studio, you'll see a few parameters being displayed in the preview:
* **Client**: 335dde47-0bc3-4dc8-8faa-c48b38b1539e
* **Deployment ID**: 1
* **Keyset URL**: /api/lti_consumer/v1/public_keysets/block-v1:OpenCraft+LTI101+2020_T2+type@lti_consumer+block@996c72b16070434098bc598bd7d6dbde
* **OIDC Callback URL**: /api/lti_consumer/v1/launch/
4. Add the tunnel url to the keyset url as it'll need to be accessed by the tool.
```
# This is <LMS_URL>/api/lti_consumer/v1/launch/<BLOCK_LOCATION>
https://647dd2e1.ngrok.io/api/lti_consumer/v1/public_keysets/block-v1:OpenCraft+LTI101+2020_T2+type@lti_consumer+block@996c72b16070434098bc598bd7d6dbde
```
5. Append the callback url to the LMS address.
```
# The block information comes from the OIDC login hint now :)
http://localhost:18000/api/lti_consumer/v1/launch/
```
6. Edit the tool you previously created and add the parameters above.
6. Publish block, go into LMS and launch LTI following the instructions on the XBlock.
8. Check that the LTI launch was successful.

**Notes:**
You can use my test tool as a reference: https://lti-ri.imsglobal.org/lti/tools/857.
